src/v/kafka/server/handlers/list_partition_reassignments.cc

```diff
@@ -0,0 +1,105 @@
+
+// Copyright 2020 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "kafka/server/handlers/list_partition_reassignments.h"
+
+#include "cluster/metadata_cache.h"
+#include "cluster/topics_frontend.h"
+#include "cluster/types.h"
+#include "kafka/protocol/errors.h"
+#include "kafka/protocol/schemata/list_partition_reassignments_request.h"
+#include "kafka/protocol/schemata/list_partition_reassignments_response.h"
+#include "kafka/server/errors.h"
+#include "kafka/server/fwd.h"
+#include "kafka/server/quota_manager.h"
+#include "model/metadata.h"
+#include "model/namespace.h"
+#include "model/timeout_clock.h"
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/sstring.hh>
+
+#include <absl/container/node_hash_map.h>
+
+#include <algorithm>
+#include <chrono>
+#include <iterator>
+#include <vector>
+
+namespace kafka {
+
+template<>
+ss::future<response_ptr> list_partition_reassignments_handler::handle(
+  request_context ctx, [[maybe_unused]] ss::smp_service_group ssg) {
+    list_partition_reassignments_request request;
+    request.decode(ctx.reader(), ctx.header().version);
+    log_request(ctx.header(), request);
+    list_partition_reassignments_response resp;
+
+    if (!request.data.topics.has_value() || request.data.topics->empty()) {
+        co_return co_await ctx.respond(std::move(resp));
+    }
+
+    resp.data.topics.reserve(request.data.topics->size());
+    auto timeout_ms = request.data.timeout_ms;
+    kafka::error_code top_level_err;
+    ss::sstring top_level_err_message;
+
+    for (const list_partition_reassignments_topics& topic :
+         *request.data.topics) {
+        ongoing_topic_reassignment topic_reassignment{.name = topic.name};
+
+        for (const int32_t p_id : topic.partition_indexes) {
+            model::ntp ntp{
+              model::ns{"kafka"}, topic.name, model::partition_id{p_id}};
```

Comment:
Again, hardcoding for the kafka namespaces makes sense to me here because the request will always be for partitions in a topic and thus always in the kafka namespace. I'm open to ideas though.